presto-main/src/main/java/com/facebook/presto/operator/PartitionedLookupSourceFactory.java

```diff
@@ -139,7 +139,13 @@ public PartitionedLookupSourceFactory(List<Type> types, List<Type> outputTypes,
     @Override
     public int partitions()
     {
-        return partitions.length;
+        lock.readLock().lock();
+        try {
+            return partitions.length;
```

Comment:
`partitions.length` is final, because `partitions` is final, so this doesn't need the lock.
Are you adding this to get a code analysis tool happy?

Reply:
I missed that it's final. Yeah it's for making IntelliJ happy, I will just add a comment and suppress the warnings.